src/main/java/io/bf2/kafka/common/Config.java

```diff
@@ -71,60 +48,10 @@ public class Config {
-            SEGMENT_INDEX_BYTES_CONFIG + ":10485760"
-    );
-
-    public static final Set<String> DEFAULT_MUTABLE_CONFIG_KEYS = Set.of(
-            CLEANUP_POLICY_CONFIG,
-            COMPRESSION_TYPE_CONFIG,
-            DELETE_RETENTION_MS_CONFIG,
-            FILE_DELETE_DELAY_MS_CONFIG,
-            FLUSH_MESSAGES_INTERVAL_CONFIG,
-            FLUSH_MS_CONFIG,
-            INDEX_INTERVAL_BYTES_CONFIG,
-            MAX_COMPACTION_LAG_MS_CONFIG,
-            MAX_MESSAGE_BYTES_CONFIG,
-            MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG,
-            MESSAGE_TIMESTAMP_TYPE_CONFIG,
-            MIN_CLEANABLE_DIRTY_RATIO_CONFIG,
-            MIN_COMPACTION_LAG_MS_CONFIG,
-            PREALLOCATE_CONFIG,
-            RETENTION_BYTES_CONFIG,
-            RETENTION_MS_CONFIG,
-            SEGMENT_BYTES_CONFIG,
-            SEGMENT_INDEX_BYTES_CONFIG,
-            SEGMENT_JITTER_MS_CONFIG,
-            SEGMENT_MS_CONFIG,
-            UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG,
-            MESSAGE_DOWNCONVERSION_ENABLE_CONFIG
-    );
-
-    public static final String DEFAULT_MAX_MESSAGE_BYTES= "1048588";
-    public static final String DEFAULT_MIN_SEGMENT_BYTES = "52428800";
-    public static final String DEFAULT_MIN_SEGMENT_MS = "600000";
-    public static final Set<String> DEFAULT_RANGE_CONFIG_SET = Set.of(
-            String.format("%s::%s", MAX_MESSAGE_BYTES_CONFIG, DEFAULT_MAX_MESSAGE_BYTES),
-            String.format("%s:%s:", SEGMENT_BYTES_CONFIG, DEFAULT_MIN_SEGMENT_BYTES),
-            String.format("%s:%s:", SEGMENT_MS_CONFIG, DEFAULT_MIN_SEGMENT_MS));
-
-    public static final String DEFAULT_CONFIG_VALUE_CONFIGS = String.join(",", DEFAULT_ENFORCED_VALUE_SET);
-    public static final String DEFAULT_MUTABLE_CONFIGS = String.join(",", DEFAULT_MUTABLE_CONFIG_KEYS);
-    public static final String DEFAULT_RANGE_CONFIGS = String.join(",", DEFAULT_RANGE_CONFIG_SET);
-
     public static final ConfigDef TOPIC_POLICY_CONFIG_DEF = new ConfigDef()
-            .define(ENFORCED_VALUE_CONFIGS, ConfigDef.Type.LIST, DEFAULT_CONFIG_VALUE_CONFIGS, ConfigDef.Importance.MEDIUM, ENFORCED_VALUE_CONFIGS_DOC)
-            .define(MUTABLE_CONFIGS, ConfigDef.Type.LIST, DEFAULT_MUTABLE_CONFIGS, ConfigDef.Importance.MEDIUM, MUTABLE_CONFIGS_DOC)
-            .define(RANGE_CONFIGS, ConfigDef.Type.LIST, DEFAULT_RANGE_CONFIGS, ConfigDef.Importance.MEDIUM, RANGE_CONFIGS_DOC)
+            .define(ENFORCED_VALUE_CONFIGS, ConfigDef.Type.LIST, Collections.emptyList(), ConfigDef.Importance.MEDIUM, ENFORCED_VALUE_CONFIGS_DOC)
+            .define(MUTABLE_CONFIGS, ConfigDef.Type.LIST, Collections.emptyList(), ConfigDef.Importance.MEDIUM, MUTABLE_CONFIGS_DOC)
+            .define(RANGE_CONFIGS, ConfigDef.Type.LIST, Collections.emptyList(), ConfigDef.Importance.MEDIUM, RANGE_CONFIGS_DOC)
             .define(TOPIC_CONFIG_POLICY_ENFORCED, ConfigDef.Type.BOOLEAN, false, ConfigDef.Importance.MEDIUM, TOPIC_CONFIG_POLICY_ENFORCED_DOC);
```

Comment:
does `false` still make sense as a plugin default? If someone is including this plugin in the broker's classpath is that not enough of a signal that it should be enforced? I like keeping the option as a way to disable it but personally I would flip the default.

Reply:
IMO, I think we have 2 options:
1. as you suggested, make `TOPIC_CONFIG_POLICY_ENFORCED` default to true, to enable it by default. But in this way, user will not be able to create topic with config change or alter topic configs, because mutable configs are empty by default. Maybe we also default set mutable configs to ALL configs?
2. make `TOPIC_CONFIG_POLICY_ENFORCED` default to false, to disable it by default. This way, when user adopt it, there will be no impact to it, until the configs are altered.

I think option (2) is easier and not complicated. WDYT?

Reply:
Hmm. I hadn't thought through the implications of the default being an empty list. I understand point 1, in that by default it will lock down everything but I still find it odd to need to add explicit config to enable the plug-in to actually function. 


I'd been keen to hear from others on this.

Reply:
Given that this project is a collection of different enforcement mechanisms (authorizer, number of partitions, and now topic configs), I think it still does provide functionality even if this is set to false by default. For now, I think false is a sane enough default.